src/languages/processing.js

```diff
@@ -6,46 +6,432 @@ Website: https://processing.org
+    "PShaderresetShader",
+    "shader",
+    "createFont",
+    "loadFont",
+    "text",
+    "textFont",
+    "textAlign",
+    "textLeading",
+    "textMode",
+    "textSize",
+    "textWidth",
+    "textAscent",
+    "textDescent",
+    "abs",
+    "ceil",
+    "constrain",
+    "dist",
+    "exp",
+    "floor",
+    "lerp",
+    "log",
+    "mag",
+    "map",
+    "max",
+    "min",
+    "norm",
+    "pow",
+    "round",
+    "sq",
+    "sqrt",
+    "acos",
+    "asin",
+    "atan",
+    "atan2",
+    "cos",
+    "degrees",
+    "radians",
+    "sin",
+    "tan",
+    "noise",
+    "noiseDetail",
+    "noiseSeed",
+    "random",
+    "randomGaussian",
+    "randomSeed"
+  ];
+  const IDENT = hljs.IDENT_RE;
+  const FUNC_NAME = {
+    variants: [
+      {
+        match: regex.concat(regex.either(...BUILT_INS), regex.lookahead(/\s*\(/)),
+        className: "built_in"
+      },
+      {
+        relevance: 0,
+        match: regex.concat(
+          /\b(?!for|if|while)/,
+          IDENT, regex.lookahead(/\s*\(/)),
+        className: "title.function"
+      }
+    ]
+  };
+  const NEW_CLASS = {
+    match: [
+      /new\s+/,
+      IDENT
+    ],
+    className: {
+      1: "keyword",
+      2: "class.title"
+    }
+  };
+  const PROPERTY = {
+    relevance: 0,
+    match: [
+      /\./,
+      IDENT
+    ],
+    className: {
+      2: "property"
+    }
+  };
+  const CLASS = {
+    variants: [
+      {
+        match: [
+          /class/,
+          /\s+/,
+          IDENT,
+          /\s+/,
+          /extends/,
+          /\s+/,
+          IDENT
+        ]
+      },
+      {
+        match: [
+          /class/,
+          /\s+/,
+          IDENT
+        ]
+      }
+    ],
+    className: {
+      1: "keyword",
+      3: "title.class",
+      5: "keyword",
+      7: "title.class.inherited"
+    }
+  };
+
+  const TYPES = [
+    "boolean",
+    "byte",
+    "char",
+    "color",
+    "double",
+    "float",
+    "int",
+    "long",
+    "short",
+  ];
+  const CLASSES = [
+    "BufferedReader",
+    "PVector",
+    "PFont",
+    "PImage",
+    "PGraphics",
+    "HashMap",
+    "String",
+    "Array",
+    "FloatDict",
+    "ArrayList",
+    "FloatList",
+    "IntDict",
+    "IntList",
+    "JSONArray",
+    "JSONObject",
+    "Object",
+    "StringDict",
+    "StringList",
+    "Table",
+    "TableRow",
+    "XML"
+  ];
+  const JAVA_KEYWORDS = [
+    "abstract",
+    "assert",
+    "break",
+    "case",
+    "catch",
+    "const",
+    "continue",
+    "default",
+    "else",
+    "enum",
+    "final",
+    "finally",
+    "for",
+    "if",
+    "import",
+    "instanceof",
+    "long",
+    "native",
+    "new",
+    "package",
+    "private",
+    "private",
+    "protected",
+    "protected",
+    "public",
+    "public",
+    "return",
+    "static",
+    "strictfp",
+    "switch",
+    "synchronized",
+    "throw",
+    "throws",
+    "transient",
+    "try",
+    "void",
+    "volatile",
+    "while"
+  ];
+
   return {
```

Comment:
this seems like both the simplest, and most verbose grammar :D

Reply:
Variants for multi-match are kind of ridiculous but these lint rules make sense most of the time for reviewing complex changes.. multiple things per line often makes a diff harder to see what changed.

For example `CLASS` here kind of really annoys me.